### ISSUE TITLE
BoundsError on Julia 0.6rc1

### ISSUE THREAD

**Issue by ohadle** [dmbates/MixedModels.jl#79]
On Julia 0.6 RC1, trying to run an example from https://github.com/dmbates/MixedModelsinJulia/blob/master/SimpleLMM.ipynb, I get this error:

```
 julia> mm1 = fit!(lmm(@formula(Y ~ 1 + (1 | G)), dyestuff))
ERROR: BoundsError: attempt to access ()
  at index [1]
Stacktrace:
 [1] macro expansion at /home/ohad/.julia/v0.6/DataArrays/src/dataarray.jl:254 [inlined]
 [2] macro expansion at /home/ohad/.julia/v0.6/Compat/src/ngenerate.jl:60 [inlined]
 [3] isna(::DataArrays.DataArray{Float64,1}) at /home/ohad/.julia/v0.6/Compat/src/ngenerate.jl:57
 [4] completecases(::DataFrames.DataFrame) at /home/ohad/.julia/v0.6/DataFrames/src/abstractdataframe/abstractdataframe.jl:471
 [5] na_omit at /home/ohad/.julia/v0.6/DataFrames/src/statsmodels/formula.jl:235 [inlined]
 [6] #ModelFrame#122(::Dict{Any,Any}, ::Type{T} where T, ::DataFrames.Terms, ::DataFrames.DataFrame) at /home/ohad/.julia/v0.6/DataFrames/src/statsmodels/formula.jl:320
 [7] #ModelFrame#125(::Array{Any,1}, ::Type{T} where T, ::DataFrames.Formula, ::DataFrames.DataFrame) at /home/ohad/.julia/v0.6/DataFrames/src/statsmodels/formula.jl:333
 [8] #lmm#32(::Array{Any,1}, ::Function, ::DataFrames.Formula, ::DataFrames.DataFrame) at /home/ohad/.julia/v0.6/MixedModels/src/pls.jl:100
 [9] lmm(::DataFrames.Formula, ::DataFrames.DataFrame) at /home/ohad/.julia/v0.6/MixedModels/src/pls.jl:100
```

**Comment by nalimilan** [comment]
Yes, that's a known problem due to the latest DataArrays release. You can work around this by going back to the previous version. See https://github.com/JuliaLang/METADATA.jl/pull/9148.

**Comment by nalimilan** [comment]
I've got a fix at https://github.com/JuliaStats/DataArrays.jl/pull/251.

**Comment by ohadle** [comment]
Works, thanks!

**Comment by nalimilan** [comment]
Should be fixed with DataArrays 0.5.1.
